src/middlewared/middlewared/plugins/auth.py

```diff
@@ -403,13 +401,17 @@ def get_token_for_shell_application(self, token_id, origin):
         }
 
     @no_auth_required
-    @accepts()
+    @accepts(Str('username'))
     @returns(Bool('two_factor_auth_enabled', description='Is `true` if 2FA is enabled'))
-    async def two_factor_auth(self):
+    async def two_factor_auth(self, username):
         """
         Returns true if two factor authorization is required for authorizing user's login.
         """
-        return (await self.middleware.call('auth.twofactor.config'))['enabled']
+        return (await self.middleware.call('auth.twofactor.config'))['enabled'] and (
+            await self.middleware.call(
+                'user.query', [['username', '=', username], ['twofactor_auth_configured', '=', True]]
```

Comment:
We'll have easier time with directory services if you got through the procedure of `user.get_user_obj` -> `pw_uid` -> `<the thing>` because AD usernames are case-insensitive and may be of form `DOMAIN\\username` or `username@DOMAIN` (both are valid and will be converted into a passwd struct via get_user_obj

Reply:
Roger that. Will make appropriate changes